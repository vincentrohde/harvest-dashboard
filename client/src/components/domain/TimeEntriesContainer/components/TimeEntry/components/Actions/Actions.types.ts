export interface ActionsProps {
    handleDelete: Function;
    toggleEdit: () => void;
}
