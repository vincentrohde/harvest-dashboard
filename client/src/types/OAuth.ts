export type oAuthType = {
    access_token: string;
    account_id: string;
    expiration_date: string;
    refresh_token: string;
};
