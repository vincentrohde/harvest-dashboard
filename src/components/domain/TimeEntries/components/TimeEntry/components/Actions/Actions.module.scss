.actions-header {
    .delete, .edit {
        display: inline;
    }
}
