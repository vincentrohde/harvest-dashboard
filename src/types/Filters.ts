export interface FiltersInterface {
    dateRange: string[] | 1;
}
