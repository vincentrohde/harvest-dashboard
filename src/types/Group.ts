export type groupsType = GroupInterface[];

export interface GroupInterface {
    id: number;
    name: string;
}