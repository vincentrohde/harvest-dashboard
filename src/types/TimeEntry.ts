import { ProjectInterface } from './Project';
import { TaskInterface } from './Task';

export type timeEntriesType = TimeEntryInterface[];

export interface ShallowTimeEntryInterface {
    hours: number | string;
    id: number;
    notes: string;
    spent_date: string;
    is_running: boolean;
    project: ProjectInterface;
    task: TaskInterface;
}

export interface TimeEntryInterface extends ShallowTimeEntryInterface {
    hours: number;
}

export interface ShallowSubmissionEntryInterface {
    notes: ShallowTimeEntryInterface['notes'];
    spent_date: ShallowTimeEntryInterface['spent_date'];
    hours: ShallowTimeEntryInterface['hours'];
    project_id: number | string;
    task_id: number | string;
}

export interface TimeEntrySubmissionInterface extends ShallowSubmissionEntryInterface {
    hours: TimeEntryInterface['hours'];
    project_id: number;
    task_id: number;
}