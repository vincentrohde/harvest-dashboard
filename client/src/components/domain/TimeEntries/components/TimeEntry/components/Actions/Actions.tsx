// Libs
import React from 'react';
import { Icon } from 'semantic-ui-react';

// Types
import { ActionsProps } from './Actions.types';

// Styles
import './Actions.scss';

const Actions = ({ handleDelete, toggleEdit }: ActionsProps) => (<div className="actions-header">
    <Icon className="delete js-delete" name="trash alternate" onClick={handleDelete}/>
    <Icon className="edit js-edit" name="pencil" onClick={toggleEdit}/>
</div>);

export default Actions;
