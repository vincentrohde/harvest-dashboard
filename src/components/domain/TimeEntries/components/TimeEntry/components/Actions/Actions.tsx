// Libs
import React from 'react';
import { Icon } from 'semantic-ui-react';
import classNames from 'classnames';

// Types
import { ActionsProps } from './Actions.types';

// Styles
import styles from './Actions.module.scss';

const Actions = ({ handleDelete, toggleEdit }: ActionsProps) => (<div className={styles['actions-header']}>
    <Icon className={classNames(styles.delete, 'js-delete')} name="trash alternate" onClick={handleDelete}/>
    <Icon className={classNames(styles.edit, 'js-edit')} name="pencil" onClick={toggleEdit}/>
</div>);

export default Actions;
