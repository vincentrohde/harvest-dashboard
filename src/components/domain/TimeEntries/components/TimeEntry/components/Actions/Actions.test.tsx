// Libs
import { render } from '@testing-library/react';
import React from 'react';
import '@testing-library/jest-dom/extend-expect';

// Components
import Actions from './Actions';

// Types
import { ActionsProps } from './Actions.types';

const mockData: ActionsProps = {
    handleDelete: () => null,
    toggleEdit: () => null
}

describe('<Actions />', () => {
    it('renders', () => {
        const { asFragment } = render(
            <Actions {...mockData} />
        );

        expect(asFragment()).toMatchSnapshot();
    });
})
