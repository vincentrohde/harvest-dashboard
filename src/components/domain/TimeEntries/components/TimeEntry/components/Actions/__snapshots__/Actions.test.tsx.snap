// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Actions /> renders 1`] = `
<DocumentFragment>
  <div
    class="actions-header"
  >
    <i
      aria-hidden="true"
      class="trash alternate icon delete js-delete"
    />
    <i
      aria-hidden="true"
      class="pencil icon edit js-edit"
    />
  </div>
</DocumentFragment>
`;
